test/test_decorators.py

```diff
@@ -0,0 +1,82 @@
+import pytest
+
+from nameko_amqp_retry.backoff import Backoff
+from nameko_amqp_retry.decorators import entrypoint_retry
+
+
+class TestEntrypointRetry(object):
+
+    class MyService:
+
+        def __init__(self, raises=None):
+            self.raises = raises
+
+        @entrypoint_retry()
+        def method1(self, arg1):
+            if self.raises:
+                raise self.raises
```

Comment:
Or possibly do away with the multiple services and pass the `raises` argument into each method as an argument. That might make things a little clearer.